src/AppBundle/Entity/Post.php

```diff
@@ -149,35 +149,38 @@ public function setContent($content)
         $this->content = $content;
     }
 
-    public function getAuthorEmail()
+    public function getPublishedAt()
     {
-        return $this->authorEmail;
+        return $this->publishedAt;
     }
 
-    /**
-     * @param string $authorEmail
-     */
-    public function setAuthorEmail($authorEmail)
+    public function setPublishedAt(\DateTime $publishedAt)
     {
-        $this->authorEmail = $authorEmail;
+        $this->publishedAt = $publishedAt;
     }
 
     /**
-     * Is the given User the author of this Post?
+     * @return User
      */
-    public function isAuthor(User $user)
+    public function getAuthor()
     {
-        return $user->getEmail() === $this->getAuthorEmail();
+        return $this->author;
     }
 
-    public function getPublishedAt()
+    /**
+     * @param User $author
+     */
+    public function setAuthor(User $author)
     {
-        return $this->publishedAt;
+        $this->author = $author;
     }
 
-    public function setPublishedAt(\DateTime $publishedAt)
+    /**
+     * Is the given User the author of this Post?
+     */
+    public function isAuthor(User $user = null)
     {
-        $this->publishedAt = $publishedAt;
+        return $user === $this->author;
```

Comment:
As Oleg said, both variables $user and $this->author can be null at the same time - I just want to prevent weird bugs in the future. I think it's unsafe just rely on `nullable` option which could be changed.